Toolchain: Fail to build gettext with GCC 14.2

Recently, I updated SerenityOS to the latest commit (b16c2906e83f269ac1b23e763c67436fdb35d0f9) and tried to rebuild the whole system and ports, but I got a weird compile error while compiling the gettext port. It says `expected 'const struct utimbuf *' but argument is of type 'const struct utimbuf *'`.

```
[gettext/build] utime.c: In function 'rpl_utime':
[gettext/build] utime.c:284:23: error: passing argument 2 of 'utime' from incompatible pointer type [-Wincompatible-pointer-types]
[gettext/build]   284 |   return utime (name, ts);
[gettext/build]       |                       ^~
[gettext/build]       |                       |
[gettext/build]       |                       const struct utimbuf *
[gettext/build] In file included from ./utime.h:34,
[gettext/build]                  from utime.c:22:
[gettext/build] /home/a/git/serenity/Userland/Libraries/LibC/utime.h:13:33: note: expected 'const struct utimbuf *' but argument is of type 'const struct utimbuf *'
[gettext/build]    13 | int utime(char const* pathname, const struct utimbuf*);
[gettext/build]       |                                 ^~~~~~~~~~~~~~~~~~~~~
```

I tried the following commands on a new Ubuntu24 VM  but the result was the same.

```
sudo apt update
sudo apt install build-essential cmake curl libmpfr-dev libmpc-dev libgmp-dev e2fsprogs ninja-build qemu-system-gui qemu-system-x86 qemu-utils ccache rsync unzip texinfo libssl-dev
sudo apt install libgtk-3-dev libpixman-1-dev libsdl2-dev libslirp-dev libspice-server-dev
sudo apt install git

git clone https://github.com/SerenityOS/serenity.git
cd serenity

./Meta/serenity.sh build

# SerenityOS works fine here
./Meta/serenity.sh run

cd ./Ports/gettext
./package.sh
```

--- a/Userland/Libraries/LibC/utime.h
+++ b/Userland/Libraries/LibC/utime.h
@@ -7,6 +7,7 @@
 #pragma once
 
 #include <sys/cdefs.h>
+#include <sys/types.h>
 
 __BEGIN_DECLS
 

